web/client/components/geostory/media/index.js

```diff
@@ -23,7 +23,10 @@ export const typesMap = {
 };
 
 const ErrorComponent = () => <div className="ms-media-error"><Glyphicon glyph="exclamation-sign"/></div>;
-const LoaderComponent = () => <div className="ms-media-loader"><Loader size={52}/></div>;
+
+// The div wrapping <Loader /> is passed a style that includes a height e.g. 500px big enough such that we're able to scroll
+// to a section during lazy loading of media. See https://github.com/geosolutions-it/MapStore2/issues/5766
+const LoaderComponent = ({ style = {}, size = 52 }) => <div style={style} className="ms-media-loader"><Loader size={size}/></div>;
```

Comment:
The issue is still visible even if we apply a height greater than the previous one because the height of media could change depending of the size property, the width of device or their own natural height.

You can see in this .gif the video is not correctly selected even with height equal to 500px
![lazy-error](https://user-images.githubusercontent.com/19175505/92367452-b2c38300-f0f6-11ea-9c7d-6125eecbf61f.gif)

This issue is related to media contents rendered inside paragraph or column of immersive section. We should try to set the right height based on the type of media to ensure a consistent transition between loading and actual component and avoid issue with the scrollTo action. We could set the height of loader using CSS similar to the actual component. If you see the DOM nodes the ms-media-loader is contained inside a specific node with type, size and alignement

eg

![image](https://user-images.githubusercontent.com/19175505/92369634-8c531700-f0f9-11ea-9fde-bb6e0eba9325.png)

- [ ] Video: it uses a [resolutions of 16/9](https://github.com/geosolutions-it/MapStore2/blob/0f5afda5d60a9962ec88954d77a6e292a5c3437e/web/client/components/geostory/media/Video.jsx#L208) so we could use [`padding-top: 56.25%;`](https://www.w3schools.com/howto/howto_css_aspect_ratio.asp) to the `ms-media-loader` class and `position: asbsolute` for the loader div child with top and left to 50% and transition to -50%, -50%. The loader should automatically adjust to 16/9 with this kind of css.

- [ ] Map: we should apply to loader component a similar CSS as for the actual component, see [paragraph](https://github.com/geosolutions-it/MapStore2/blob/0f5afda5d60a9962ec88954d77a6e292a5c3437e/web/client/themes/default/less/geostory.less#L621-L630) and [immersive](https://github.com/geosolutions-it/MapStore2/blob/0f5afda5d60a9962ec88954d77a6e292a5c3437e/web/client/themes/default/less/geostory.less#L735-L744) css of map contents

- [ ] Web page: we should apply to loader component a similar CSS as for the actual component, see [paragraph](https://github.com/geosolutions-it/MapStore2/blob/0f5afda5d60a9962ec88954d77a6e292a5c3437e/web/client/themes/default/less/geostory.less#L639-L645) and css [immersive](https://github.com/geosolutions-it/MapStore2/blob/0f5afda5d60a9962ec88954d77a6e292a5c3437e/web/client/themes/default/less/geostory.less#L754-L760) of web page contents. (I'm not sure if this needs changes to verify)

Exception

- [ ] Image: the height of images is computed by the browser based on the naturalHeight of the original source and we have not this information before loading. @offtherailz Should we need to store the naturalHeight and naturalWidth of an image in the story resources after adding or updating the src of an image in the media editor? If so we could calculate the height of the loader based on the resolution of original image. This approach needs inline style with the padding approach used by video using the resolution of the original image.


At the end we should not notice differences on the height before and after loading, eg: video media

![image](https://user-images.githubusercontent.com/19175505/92367922-557c0180-f0f7-11ea-8d3d-e3b194183f9c.png)

![image](https://user-images.githubusercontent.com/19175505/92367257-7a23a980-f0f6-11ea-8119-8b583289a114.png)


Important: let's start from video content and see if the issue 5 described here is solved https://github.com/geosolutions-it/MapStore2/issues/5757 before to apply changes to every media contents

Reply:
As discussed we should save the size information for image, and we can post-pone saving the resolution for video.